contrib/platform/test/com/sun/jna/platform/mac/SystemBTest.java

```diff
@@ -64,33 +64,39 @@ public void testSysctl() {
         final String mibName = "hw.logicalcpu";
         final int nCpu = Runtime.getRuntime().availableProcessors();
 
-        IntByReference size = new IntByReference(SystemB.INT_SIZE);
-        Pointer p = new Memory(size.getValue());
-        int ret = SystemB.INSTANCE.sysctlbyname(mibName, p, size, null, 0);
-        assertEquals(ret, 0);
+        // This sysctl returns a 32-bit integer cpu count
+        size_t size = new size_t(Integer.BYTES);
+        Pointer p = new Memory(size.longValue());
+        size_t.ByReference plen = new size_t.ByReference(size);
+        assertEquals(0, SystemB.INSTANCE.sysctlbyname(mibName, p, plen, null, size_t.ZERO));
         // These values should be equal unless affinity is set, limiting nCpu
         assertTrue(p.getInt(0) >= nCpu);
 
-        size = new IntByReference();
-        ret = SystemB.INSTANCE.sysctlnametomib(mibName, null, size);
-        assertEquals(ret, 0);
-        // Size should be 2
-        assertEquals(size.getValue(), 2);
-
-        Pointer mibp = new Memory(size.getValue() * SystemB.INT_SIZE);
-        ret = SystemB.INSTANCE.sysctlnametomib(mibName, mibp, size);
-        assertEquals(ret, 0);
-        // Size should be 2
-        assertEquals(size.getValue(), 2);
-
-        int[] mib = mibp.getIntArray(0, size.getValue());
+        // Get the same value by converting the string to the MIB array
+        // Get the size of the MIB array
+        size = new size_t(Integer.BYTES);
+        size_t.ByReference sizep = new size_t.ByReference(size);
+        assertEquals(0, SystemB.INSTANCE.sysctlnametomib(mibName, null, sizep));
+        // Array size should be 2
+        int sizepval = sizep.getValue().intValue();
+        assertEquals(2, sizepval);
+
+        // Allocate the correct size and fill the MIB array
+        Pointer mibp = new Memory(sizepval * SystemB.INT_SIZE);
+        assertEquals(0, SystemB.INSTANCE.sysctlnametomib(mibName, mibp, sizep));
+        // Array size should still be 2
+        sizepval = sizep.getValue().intValue();
+        assertEquals(2, sizepval);
+
+        int[] mib = mibp.getIntArray(0, sizepval);
         // mib should be { 6, 103(?) }
         assertEquals(mib.length, 2);
         assertEquals(mib[0], 6);
 
-        size = new IntByReference(SystemB.INT_SIZE);
-        p = new Memory(size.getValue());
-        ret = SystemB.INSTANCE.sysctl(mib, mib.length, p, size, null, 0);
+        size = new size_t(Integer.BYTES);
+        p = new Memory(size.longValue());
+        plen = new size_t.ByReference(size);
```

Comment:
I wrote above, that I would add java getter and setter, this is the location where it matters. This can be easier written as (dropping line 96). Declare `p` as Memory and then:

```java
p = new Memory(Integer.BYTES);
plen = new size_t.ByReference(p.size());
```

That way the connection between `p` and `plen` becomes obvious.

Reply:
I actually did already have a constructor that takes a `long` for this particular use case, but I agree having the primitve type getters/setters makes things simpler overall.